gateway/scim_ble.py

```diff
@@ -7,146 +7,87 @@
+    # if ble is added in update, just add it.
+    if not entry:
+        entry = ble_create_device(request,request.json["id"])
+        session.commit()
+        return
 
-    user.id = request.json.get("id")
-    user.device_display_name = request.json.get("deviceDisplayName")
-    user.admin_state = request.json.get("adminState")
-    user.version_support = request.json[
-        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "versionSupport")
-    user.device_mac_address = request.json[
-        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "deviceMacAddress")
-    user.is_random = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "isRandom")
-    user.pairing_methods = request.json[
-        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
-        "urn:ietf:params:scim:schemas:extension:pairingNull:2.0:Device")
-    user.pairing_null = request.json[
-        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"].get(
+    ble_json=request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"]
+    entry.version_support = ble_json.get("versionSupport")
+    entry.device_mac_address = ble_json.get("deviceMacAddress")
+    entry.is_random = ble_json.get("isRandom")
+    entry.pairing_methods = ble_json.get("pairingMethods")
+    entry.pairing_null = ble_json.get(
         "urn:ietf:params:scim:schemas:extension:pairingNull:2.0:Device")
-    user.pairing_just_works_key = request.json[
-        "urn:ietf:params:scim:schemas:extension:ble:2.0:Device"][
-        "urn:ietf:params:scim:schemas:extension:pairingJustWorks:2.0:Device"].get("key")
-    user.pairing_pass_key = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"][
-        "urn:ietf:params:scim:schemas:extension:pairingPassKey:2.0:Device"].get("key")
-    user.pairing_oob_key = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"][
-        "urn:ietf:params:scim:schemas:extension:pairingOOB:2.0:Device"].get("key")
-    user.pairing_oobrn = request.json["urn:ietf:params:scim:schemas:extension:ble:2.0:Device"][
-        "urn:ietf:params:scim:schemas:extension:pairingOOB:2.0:Device"].get("randNumber")
-    user.modified_time = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+    entry.pairing_just_works_key = ble_json.get(
+        "urn:ietf:params:scim:schemas:extension:pairingJustWorks:2.0:Device").get("key")
+    entry.pairing_pass_key = ble_json.get(
+        "urn:ietf:params:scim:schemas:extension:pairingPassKey:2.0:Device").get("key")
+    entry.pairing_oob_key = ble_json.get(
+        "urn:ietf:params:scim:schemas:extension:pairingOOB:2.0:Device").get("key")
+    entry.pairing_oobrn = ble_json.get(
+        "urn:ietf:params:scim:schemas:extension:pairingOOB:2.0:Device").get("randNumber")
 
     session.commit()
-    return make_response(jsonify(user.serialize()), 200)
+    return
```

Comment:
Probably shouldn't commit here. 
```suggestion
    return entry
```

Reply:
Right